### PR TITLE
Fixed minor mistakes, enabled full bulkification.

### PR DESCRIPTION
**Pull request by brianmfear**

### CODE REVIEW

**Comment by dschach** [comment]
Thanks, @brianmfear - good catch on passing collections to invocable methods.

@codefriar do you want to take a look as well? It looks good to me.

**Comment by brianmfear** [comment]
@dschach You're welcome! Anything I can do to help make the project better.

**Comment by codefriar** [comment]
@brianmfear, Thank you for this. The bulkification is fantastic. There are a couple of little things I want to change, but nothing serious enough to merit asking you to do it. Things like, I want to write a bit more on a couple of method header comments. Thank you for this. Do you want to collaborate and cross post a blog post about this. As you said, I think this is pretty epic.

**Review by codefriar — APPROVED** [review]
Epic.

**Comment by brianmfear** [comment]
@codefriar If you're up for it, sure. I haven't done a post for years 😅

**Comment by codefriar** [comment]
Let’s do it. I’ll spin up a notion doc and invite you

File: force-app/main/default/classes/Universal Flow Invocable/Invocable.cls [hunk 1]
```diff
@@ -126,109 +119,96 @@ public abstract class Invocable implements Callable {
 		 */
 		public Map<String, Object> toCallableParamMap() {
 			Map<String, Object> paramMap = new Map<String, Object>();
+			if (parameters == null) {
+				return paramMap;
+			}
 			for (UniversalFlowInputOutputParameter parameter : parameters) {
-				switch on parameter.parameterType.toLowerCase() {
```
Comment (dschach) [inline force-app/main/default/classes/Universal Flow Invocable/Invocable.cls:130]: Out of curiosity, @brianmfear, why did you move from switch (fast) to a series of ifs (slow, unless you use else if)?

File: force-app/main/default/classes/Universal Flow Invocable/Invocable.cls [hunk 2]
```diff
@@ -126,109 +119,96 @@ public abstract class Invocable implements Callable {
...
-								' for parameter ' +
-								parameter.parameterName
-						);
-					}
+				Object value;
+
+				if (parameter.booleanValue != null) {
+					value = parameter.booleanValue;
+				}
+				if (parameter.booleans != null) {
```
Comment (dschach) [inline force-app/main/default/classes/Universal Flow Invocable/Invocable.cls:131]: @brianmfear Would an else if be faster than just if? That way the code skips as soon as it gets a match.

Comment (codefriar) [reply force-app/main/default/classes/Universal Flow Invocable/Invocable.cls:131]: @dschach i'll let @brianmfear definitively answer, but from my conversation with him yesterday / today the switch to InstanceOf rather than the exception hack I was using means that we could end up in a situation where we have a return value that is both, for instance, a decimal and a long. In situations like that, this if structure would allow all of the applicable output variables to be populated, so the resulting flow developer can make use of whichever version they want.

Comment (brianmfear) [reply force-app/main/default/classes/Universal Flow Invocable/Invocable.cls:131]: Technically, I think you'd save an average of probably ~0.005ms per parameter with all the elses in place. This code, without the elses, actually runs faster than the switch statement that used to be there, by a lot. It would take hundreds of parameters to add up to even a few ms of wasted time. The other thought was to be able to make it so that you can send one parameter with every value set to get 100% coverage at once.

Comment (brianmfear) [reply force-app/main/default/classes/Universal Flow Invocable/Invocable.cls:131]: That said, I'm not opposed to elses in general, I just felt they weren't necessary here. After all, if someone supplies multiple values in a single parameter, this code would operate just as arbitrarily as one that took a shortcut. I guess it might even make more sense to add guardrails here by throwing an error if multiple parameters were passed in by accident.

Comment (brianmfear) [reply force-app/main/default/classes/Universal Flow Invocable/Invocable.cls:131]: @codefriar ... @dschach was talking about Invocable, not UniversalFlowInputOutputParameter. I use the technique in UniversalFlowInputOutputParameter to allow multiple outputs, since the polyfill method was misbehaving (I kept getting a NullPointerException randomly), so we populate as many legal values as we can. For the Invocable method, it could have been else-if all the way through. It's a harmless microoptimization, but does force one to write more unit tests to get 100% coverage. Either way is pretty insignificant in the grand scheme of things.
